torch/csrc/jit/backends/backend_detail.cpp

```diff
@@ -147,6 +158,15 @@ Module codegen_backend_module(
   loweredModule.define(
       create_backend_ct.format(create_backend_te), loweredModuleResolver());
 
+  // Helper function to expose backend.is_available() to Module generation code.
+  loweredModule.define(
+      R"(
+            def __is_available(self):
+                self.__create_backend()
```

Comment:
`self.__create_backend()` is called in `__setstate__`. Would this creates another backend instance?

Reply:
Yes, I need to expose it such that an instance of the backend is available --just as the old "preprocess" code needed to do this.
Actually just as the old code, there were +1 instance of backend created because indeed a call to self.__create_backend() is necessary to make sure self.__backend exists.

In the ideal case self.__create_backend() will handle the lazy instantiation, but cannot figure out a way to check whether self.__backend has been already instantiated given it's created with a instance of a capsule.

Can you expand more on how optional would work?

The other option I can think of is to add a boolean field just to tag whether self.__backend has been instantiated already.

Reply:
> Can you expand more on how optional would work?

Try setting the type of the `__backend` attribute to `Optional[TBackend]` and initializing it to `None` and then checking that it is `None` before creating it in `__create_backend`.

Reply:
Implemented with Optional value. Thanks @SplitInfinity 
Tomorrow I'll implement it with a boolean flag to get a sense of which path is cleaner.

Reply:
Ok, so reverted the implementation to not use optional value, and instead we just make sure that the backend instance is created once on __setstate__ for the regular save/load flow, and make a separate call to __create_backend at LoweredModule creation time (so we can call __is_available()).

The reason is two fold:
* This is just much cleaner.
* The mobile interpreter does not support get attribute for things that are not tensors (that cause the CI failure --thanks @iseeyuan for helping debug that). This also prevents implementing the other solution of adding a boolean attribute as flag to check for backend initialization.

Reply:
Wait, wasn't whole point to try to create the backend only once? But right now it's still created twice during lowering?

Reply:
That’s right. In the case where the LoweredModule is created there’s still 2 backend instances being created, so we can call ins_available().

The issue is that there’s no way to make the optional argument work for mobile; and adding a bool argument to check for initialization has the same issue: from @iseeyuan ”GET_ATTR only works with parameters (tensors) using slot. Since there’s no source info (to save binary size of source parser), it does not work with other types of attributes.”

Having the extra instance of check is_available() is not so bad since is_available() is pretty much a const function.

The arguably more important flow when the LoweredModule is loaded only creates a single backend instance (at __setstate__ time).

Reply:
Ok, I implemented a way to only have 1 instance on any case (I validated that's the case).

Please take a look.